src/Omega_h_amr_transfer.cpp

```diff
@@ -24,4 +27,36 @@ void amr_transfer_linear_interp(Mesh* old_mesh, Mesh* new_mesh,
   }
 }
 
+static void amr_transfer_levels(Mesh* old_mesh, Mesh* new_mesh,
+    Int prod_dim, LOs same_ents2old_ents, LOs same_ents2new_ents,
+    Few<LOs, 4> mods2mds, LOs prods2new_ents) {
+  auto levels = old_mesh->ask_levels(prod_dim);
+  auto tag_data = old_mesh->get_tag<Byte>(prod_dim, "level");
+  auto new_data = Write<Byte>(new_mesh->nents(prod_dim), -1);
+  Int offset = 0;
+  for (Int mod_dim = max2(Int(EDGE), prod_dim); mod_dim <= old_mesh->dim();
+      ++mod_dim) {
+    auto nprods_per_mod = hypercube_split_degree(mod_dim, prod_dim);
+    auto nmods_of_dim = mods2mds[mod_dim].size();
+    for (LO i = 0; i < mods2mds[mod_dim].size(); ++i) {
+      auto old_idx = mods2mds[mod_dim][i];
+      auto old_level = levels[old_idx];
+      for (Int j = 0; j < nprods_per_mod; ++j) {
+        auto new_idx = prods2new_ents[offset + (i * nprods_per_mod + j)];
+        new_data[new_idx] = old_level + 1;
+      }
+    }
+    offset += nprods_per_mod * nmods_of_dim;
+  }
+  transfer_common2(old_mesh, new_mesh, prod_dim, same_ents2old_ents,
+      same_ents2new_ents, tag_data, new_data);
+}
+
+void amr_transfer_leaves_and_levels(Mesh* old_mesh, Mesh* new_mesh,
+    Int prod_dim, LOs same_ents2old_ents, LOs same_ents2new_ents,
+    Few<LOs, 4> mods2mds, LOs prods2new_ents) {
+  amr_transfer_levels(old_mesh, new_mesh, prod_dim, same_ents2old_ents,
+      same_ents2new_ents, mods2mds, prods2new_ents);
+}
```

Comment:
For transferring leaves, it would be really convenient if there was some notion of mods2new_ents. I'm looking through `Omega_h_modify.cpp` to see if there are any hints. Any advice on how I might build that?

Reply:
That is prods2new_ents, its just that there are `nchild` prods per mod, and mods come from multiple dimensions...

Reply:
I want to know the new index of an entity that was marked for refinement (a parent), but was kept around because of the flag `keep_mods`.  As far as I can tell, `prods2new_ents` only maps the number of newly created entities (children). Is the parent entity always just indexed one before all of the child entities that are obtained with the first child entity from `prods2new_ents`?

Reply:
aaah okay. Thats a combination of `mods2mds` and `same_ents2old_ents` and `same_ents2new_ents`... First you combine the latter two, and I think if thats precomputed before I would have called it `old_ents2new_ents`. Then you can do
```cpp
auto mods2new_ents = unmap(mods2mds, old_ents2new_ents);
```
I can dig up the `old_ents2new_ents` computation if you don't find it elsewhere in the code.